server/channels/app/post_metadata.go

```diff
@@ -362,9 +362,14 @@ func (a *App) getImagesForPost(c request.CTX, post *model.Post, imageURLs []stri
 	}
 
 	for _, imageURL := range imageURLs {
-		// MM-51060 - don't accept Mattermost permalinks as Open Graph Image URLs.
-		if isPermalink := looksLikeAPermalink(imageURL, a.GetSiteURL()); isPermalink {
-			continue
+		// prevent infinite loop if a OG image URL is the same post's permalink
+		resolvedURL := resolveMetadataURL(imageURL, a.GetSiteURL())
+		isPermalink := looksLikeAPermalink(resolvedURL, a.GetSiteURL())
+		if isPermalink {
+			referencedPostID := resolvedURL[len(resolvedURL)-26:]
+			if referencedPostID == post.Id {
```

Comment:
What if the permalink in this image URL points to a different post, which then points to another permalink in an OG image URL, which points back to this post? We need to break loops even if they're longer than a single post.

I don't see why we'd need to allow any permalinks here, since a permalink isn't an image anyway and will never work. We should also never actually see permalinks here when dealing with legitimate websites.

Reply:
Hmm, true. By disabling permalink altogether we're breaking any crisscross of posts and links. Done.